### PR TITLE
Readme Input<T> documentation

### PR DESCRIPTION
**Pull request by dbart01**
### What this does

Update the code snippets to use `Input<T>` and adds a case study example showcasing where it's necessary in customer updates.

### CODE REVIEW

**Comment by dbart01** [comment]
Hey @rorytanner, would really appreciate some 👀  on the latter part of this README - the `Customer Update` section.

File: README.md [hunk 1]
```diff
@@ -1486,6 +1487,56 @@ let query = Storefront.buildQuery { $0
 }
 ```
 
+#### Customer Update [⤴](#table-of-contents)
+
+Input objects, like `Storefront.MailingAddressInput`, use `Input<T>` type to represent optional fields and distinguish `nil` values from `undefined` values.
```
Comment (rorytanner) [inline README.md]: I'm not sure about the phrasing at "Input objects, like `Storefront.MailingAddressInput`, **use** `Input<T>` **type** to represent optional fields..." -- does that work for describing `Input<T>` OR should it be something like "...use **the** `Input<T>` type to represent..." (or "...use `Input<T>` **values** to represent..."?)

Comment (dbart01) [reply README.md]: Well, `Input` is a type that represents the values contained in `Storefront.MailingAddressInput`. I'm almost thinking something like:

> Input objects, like `Storefront.MailingAddressInput`, use values of type `Input<T>` (where `T` is the type of value) to represent optional fields and distinguish `nil` values from `undefined` values (eg. `phone: Input<String>`)

What do you think?

Comment (rorytanner) [reply README.md]: @dbart01 does this work?

> Input objects, like `Storefront.MailingAddressInput`, use `Input<T>` (where `T` is the type of value) to represent optional fields and distinguish `nil` values from `undefined` values (eg. `phone: Input<String>`).

Comment (dbart01) [reply README.md]: Perfect 👍

File: README.md [hunk 2]
```diff
@@ -1486,6 +1487,56 @@ let query = Storefront.buildQuery { $0
 }
 ```
 
+#### Customer Update [⤴](#table-of-contents)
+
+Input objects, like `Storefront.MailingAddressInput`, use `Input<T>` (where `T` is the type of value) to represent optional fields and distinguish `nil` values from `undefined` values (eg. `phone: Input<String>`).
+
+The following example uses `Storefront.CustomerUpdateInput` to show how to update a customer's phone number:
```
Comment (sav007) [inline README.md]: should it state `to show how to update a JUST customer's phone number`

**Pull request merged**